### PR TITLE
Access authorization timeout

### LINKED ISSUE #1867: Support timeout settings on amp access endpoints
It would be useful to be able to specify the maxium time an authorization request should take before aborting, in case that the authorization endpoint is running slowly for whatever reason. Many publishers would prefer to serve the content (fail open) than wait for a long timeout in this scenario.

This timeout might be different for a pre-fetch (longer), which makes this difficult to control on the server (eg. using a gateway timeout).

### PR DESCRIPTION
**Pull request by dvoytenko**
Closes #1867.

### CODE REVIEW

File: extensions/amp-access/amp-access-spec.md [hunk 1]
```diff
@@ -251,14 +254,17 @@ This RPC may be called in the prerendering phase and thus it should not be used
...
  2. When evaluating ```<template>``` templates such as ```amp-mustache```.
+ 3. When providing additional variables to pingback and login URLs using `AUTHDATA(field)`.
 
 Authorization endpoint is called by AMP Runtime as a credentialed CORS endpoint. As such, it must implement CORS protocol. It should use CORS Origin and source origin to restrict the access to this service as described in the [CORS Origin Security][9]. This endpoint may use publisher cookies for its needs. For instance, it can associate the binding between the Reader ID and the Publisher’s own user identity. AMP itself does not need to know about this (and prefers not to). Reader more on [AMP Reader ID][2] and [AMP Access and Cookies][11] for more detail.
 
 AMP Runtime (or rather browser) observes cache response headers when calling Authorization endpoint. Thus the cached responses can be reused. It may or may not be desirable. If it is not desirable, the Publisher can user the appropriate cache control headers and/or RANDOM variable substitution for the endpoint URL.
 
+Authorization request is automatically timed out and assumed to have failed after 4 seconds.
```
Comment (cramforce) [inline extensions/amp-access/amp-access-spec.md]: What happens in the failure case?

Comment (dvoytenko) [inline extensions/amp-access/amp-access-spec.md]: Working on that part here: https://github.com/ampproject/amphtml/pull/1957

Comment (dvoytenko) [inline extensions/amp-access/amp-access-spec.md]: I merged some part from #1957 on failing behavior. Will reconcile the rest as part of the #1957

**Comment by dvoytenko** [comment]
Switched timeout to 3 seconds.

**Comment by cramforce** [comment]
LGTM

**Pull request merged**